### PR TITLE
Add Azure Pipelines to CI targets that support -Werror

### PR DESCRIPTION
**Pull request by rgetz**
So we find more things faster.

Signed-off-by: Robin Getz <robin.getz@analog.com>

### CODE REVIEW

**Comment by rgetz** [comment]
github is smart enough to not run CI on draft PRs, so I can't see what needs to be fixed until I make this a real pull request. I expect some to fail, and then will need to resolve issues that have slipped in since this was lost during the Travis-CI to Azure Mirgration.

**Comment by rgetz** [comment]
looks like host env vars need to be transferred to the docker image and loaded. We did that in Travis-CI, but not Azure (yet).

**Comment by pcercuei** [comment]
What's the state of this PR? I believe it's not ready with all the message() in there?

I think you can remove the checks for Appveyor and Travis, now that we don't use these anymore. That should simplify the first commit a bit.

Second commit looks OK.

**Comment by rgetz** [comment]
yes - it's not ready to commit yet. I need some more info from @tfcollins

**Comment by ccraluca** [comment]
For PowerShell scripts, can be added `$ErrorActionPreference = "Stop"` for CI/publish_deps.ps1 too? And also it is possible to add `$ErrorView = "NormalView"` after each `$ErrorActionPreference = "Stop"`? for the error message to be displayed more clearly in log

**Comment by rgetz** [comment]
> can be added `$ErrorActionPreference = "Stop"` for CI/publish_deps.ps1 too?

No problem - since that was something I wasn't testing - I didn't want to change it :)

Updated now

-Robin

**Comment by rgetz** [comment]
Ok - I think things are ready for a review/feedback

**Review by tfcollins — APPROVED** [review]

**Comment by rgetz** [comment]
Just in case someone is interested, it is turning on `-Werror` on all builds now.

```
LinuxBuilds centos_7_x86_64 : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
LinuxBuilds debian_buster_arm32v7 : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
LinuxBuilds debian_buster_arm64v8 : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
LinuxBuilds doxygen : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
LinuxBuilds ubuntu_16_04_x86_64 : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
LinuxBuilds ubuntu_18_04_x86_64 : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
LinuxBuilds ubuntu_20_04_x86_64 : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
WindowsBuilds VS2019_Win32 : Running in an Azure or AppVeyor environment, setting -Werror
WindowsBuilds VS2019_Win64 : Running in an Azure or AppVeyor environment, setting -Werror
macOSBuilds macOS_10_15 : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
macOSBuilds macOS_11 : Running in CI environment (Azure, Travis or AppVeyor), setting -Werror
```

**Review by ccraluca — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -180,10 +184,16 @@ elseif (CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
 		SET(CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -fprofile-arcs -ftest-coverage")
 		SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS}  -fprofile-arcs -ftest-coverage")
 	endif()
-	if(DEFINED ENV{CI} AND (DEFINED ENV{TRAVIS} OR DEFINED ENV{APPVEYOR}))
-		message(STATUS "Running in CI environment (Travis or AppVeyor), setting -Werror")
+	if(DEFINED ENV{TF_BUILD} OR (DEFINED ENV{CI} AND (DEFINED ENV{TRAVIS} OR DEFINED ENV{APPVEYOR})))
+		message(STATUS "Running in CI environment (Azure, Travis or AppVeyor), setting -Werror")
 		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+		if (CMAKE_C_COMPILER_ID STREQUAL "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_EQUAL "11.0")
+			set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-missing-braces")
```
Comment (pcercuei) [inline CMakeLists.txt:191]: This is new, and not talked about in the commit message, what's that for?

Comment (rgetz) [reply CMakeLists.txt:191]: yeah, this was for older versions of clang. 

In gcc - this warning was turned off in 4.8.0 (2005)

I will add something in the commit message, or better yet - split it up into a separate commit.

Comment (pcercuei) [reply CMakeLists.txt:191]: If we only enable -Werror for CI, and this a workaround for an old version of Clang, do we still need it?

Comment (rgetz) [reply CMakeLists.txt]: looks like it was only needed for the version of MacOS, that we removed...

So - it's gone now.

**Review by pcercuei — APPROVED** [review]

**Pull request merged**